File: decompiled/sources/freemarker/template/TemplateDateModel.java
package freemarker.template;

import java.util.Arrays;
import java.util.Collections;
import java.util.Date;
import java.util.List;

public interface TemplateDateModel extends TemplateModel {
    public static final int DATE = 2;
    public static final int DATETIME = 3;
    public static final int TIME = 1;
    public static final List TYPE_NAMES = Collections.unmodifiableList(Arrays.asList(new String[]{"UNKNOWN", "TIME", "DATE", "DATETIME"}));
    public static final int UNKNOWN = 0;

    Date getAsDate() throws TemplateModelException;

    int getDateType();
}

File: decompiled/sources/freemarker/template/TemplateDateModel.kt
package freemarker.template

import java.util.Arrays
import java.util.Collections
import java.util.Date

interface TemplateDateModel : TemplateModel {
    @Throws(TemplateModelException::class)
    fun getAsDate(): Date

    fun getDateType(): Int

    companion object {
        @JvmField
        val UNKNOWN: Int = 0

        @JvmField
        val TIME: Int = 1

        @JvmField
        val DATE: Int = 2

        @JvmField
        val DATETIME: Int = 3

        @JvmField
        val TYPE_NAMES: List<String> = Collections.unmodifiableList(
            Arrays.asList("UNKNOWN", "TIME", "DATE", "DATETIME"),
        )
    }
}
